src/socket.rs

```diff
@@ -861,6 +861,41 @@ impl Socket {
         }
     }
 
+    /// IP_TRANSPARENT (since Linux 2.6.24)
+    /// Setting this boolean option enables transparent proxying
+    /// on this socket.  This socket option allows the calling
+    /// application to bind to a nonlocal IP address and operate
+    /// both as a client and a server with the foreign address as
+    /// the local endpoint.  NOTE: this requires that routing be
+    /// set up in a way that packets going to the foreign address
+    /// are routed through the TProxy box (i.e., the system
+    /// hosting the application that employs the IP_TRANSPARENT
+    /// socket option).  Enabling this socket option requires
+    /// superuser privileges (the CAP_NET_ADMIN capability).
+    ///
+    /// TProxy redirection with the iptables TPROXY target also
+    /// requires that this option be set on the redirected socket.
+    /// this feature is only available on linux
+    #[cfg(any(target_os = "linux"))]
```

Comment:
Since this is not available on all OSs it needs to be behind the `all` feature. It requires `cfg` attribute, see https://github.com/rust-lang/socket2/blob/eaa5f8f66233df701983ec073f1c5683019e442e/src/socket.rs#L1333-L1345